libbeat/processors/actions/rename_fields.go

```diff
@@ -0,0 +1,69 @@
+package actions
+
+import (
+	"fmt"
+
+	"github.com/elastic/beats/libbeat/beat"
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/logp"
+	"github.com/elastic/beats/libbeat/processors"
+)
+
+type renameFields struct {
+	Fields []FieldTuple
+}
+
+type FieldTuple struct {
+	From string `config:"from"`
+	To   string `config:"to"`
+}
+
+func init() {
+	processors.RegisterPlugin("rename_fields",
+		configChecked(newRenameFields,
+			requireFields("fields")))
+}
+
+func newRenameFields(c *common.Config) (processors.Processor, error) {
+	config := struct {
+		Fields []FieldTuple `config:"fields"`
+	}{}
+	err := c.Unpack(&config)
+	if err != nil {
+		return nil, fmt.Errorf("fail to unpack the rename_fields configuration: %s", err)
+	}
+
+	// TODO: Check if some fields are not allowed to be renamed
+
+	f := &renameFields{Fields: config.Fields}
+	return f, nil
+}
+
+func (f *renameFields) Run(event *beat.Event) (*beat.Event, error) {
+
+	for _, field := range f.Fields {
+		value, err := event.Fields.GetValue(field.From)
+		if err != nil {
+			logp.Err("Could not fetch from key: %s", field.From)
+			continue
+		}
+
+		_, err = event.Fields.Put(field.To, value)
+		if err != nil {
+			logp.Err("Could not put value: %s, %v, %+v", field.To, value, err)
+			continue
+		}
+
+		err = event.Fields.Delete(field.From)
+		if err != nil {
+			logp.Err("Could not delete key: %s,  %+v", field.From, err)
+			continue
+		}
+	}
+
+	return event, nil
+}
```

Comment:
In the context of parsing multiple JSON document, I think that logging the error everytime could be quite noisy. I think we should add `ignore_missing`, this would be similar to what ingest provide if the key doesn't exist in multiple processors. See the [rename processor](https://www.elastic.co/guide/en/elasticsearch/reference/master/rename-processor.html)

Reply:
I  see we diverge from the `ignore_missing` semantic from the ingest pipeline, in their case if the flag is set to false its a fatal error. This might be OK in our usecase.

Reply:
What happens in the case of a fatal error? The original event is stored or the processed one until the error happened? We could have an additional config ignore_errors.

Reply:
@ruflin Good question, I do not have good answer, Do we have any other existing processor that implement this kind of behavior?